test/Makefile.am

```diff
@@ -33,10 +33,12 @@ check_PROGRAMS += pde_unit
 if CXX14_ENABLED
   check_PROGRAMS += dualnamedarray_unit
   check_PROGRAMS += namedindexarray_unit
+  export HAVE_CXX14=yes
```

Comment:
These lines don't have to be in `AM_TESTS_ENVIRONMENT`?

Reply:
So this did indeed work. These exports got placed under a `.PRECIOUS` target, which from what I read is meant to prevent its dependencies from getting deleted; no mention in the documentation I found about its use in exporting environment variables! Anyway after some struggles, I was able to get testing with `AM_TESTS_ENVIRONMENT` to work, so I think this is finally ready to go